engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EnsoProjectNode.java

```diff
@@ -0,0 +1,135 @@
+package org.enso.interpreter.node.expression.builtin.meta;
+
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.Truffle;
+import com.oracle.truffle.api.TruffleFile;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.Node;
+import java.util.Optional;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.node.EnsoRootNode;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.callable.atom.Atom;
+import org.enso.interpreter.runtime.data.EnsoFile;
+import org.enso.interpreter.runtime.data.Type;
+import org.enso.interpreter.runtime.error.DataflowError;
+import org.enso.interpreter.runtime.library.dispatch.TypesLibrary;
+import org.enso.pkg.Package;
+
+@BuiltinMethod(
+    type = "Project_Description",
+    name = "enso_project_builtin",
+    description = "Returns the project description of the project given as the argument")
+public abstract class EnsoProjectNode extends Node {
```

Comment:
This node is inserted into AST of the module that uses the builtin. E.g. there are different instances of this node in different sources and source locations.